plutus-core/plutus-core/src/PlutusCore/Core/Plated.hs

```diff
@@ -30,83 +35,170 @@ infixr 6 <^>
 (<^>) :: Fold s a -> Fold s a -> Fold s a
 (f1 <^> f2) g s = f1 g s *> f2 g s
 
+kindSubkinds :: Traversal' (Kind ann) (Kind ann)
+kindSubkinds f kind0 = case kind0 of
+    KindArrow ann dom cod -> KindArrow ann <$> f dom <*> f cod
+    Type{}                -> pure kind0
+
+kindSubkindsDeep :: Fold (Kind ann) (Kind ann)
+kindSubkindsDeep = cosmosOf kindSubkinds
+
+{-# INLINE tyVarDeclSubkinds #-}
+-- | Get all the direct child 'Kind's of the given 'TyVarDecl'.
+tyVarDeclSubkinds :: Traversal' (TyVarDecl tyname a) (Kind a)
+tyVarDeclSubkinds f (TyVarDecl a ty k) = TyVarDecl a ty <$> f k
+
 -- | Get all the direct child 'tyname a's of the given 'Type' from binders.
 typeTyBinds :: Traversal' (Type tyname uni ann) tyname
-typeTyBinds f = \case
+typeTyBinds f ty0 = case ty0 of
     TyForall ann tn k ty -> f tn <&> \tn' -> TyForall ann tn' k ty
     TyLam ann tn k ty    -> f tn <&> \tn' -> TyLam ann tn' k ty
-    x                    -> pure x
+    TyApp{}              -> pure ty0
+    TyIFix{}             -> pure ty0
+    TyFun{}              -> pure ty0
+    TyBuiltin{}          -> pure ty0
+    TyVar{}              -> pure ty0
```

Comment:
While I was here, I also made these definitions safer by explicitly matching on all the constructors. Was unable to find courage to do the same for `plutus-ir`.